sharkfuser/benchmarks/driver.cpp

```diff
@@ -0,0 +1,156 @@
+                                 int64_t j) {
+#ifdef FUSILLI_ENABLE_AMDGPU
+  Handle handle = FUSILLI_TRY(Handle::create(Backend::GFX942));
+#else
+  Handle handle = FUSILLI_TRY(Handle::create(Backend::CPU));
+#endif
+
+  // Build graph for the given handle (device), validate and compile it.
+  auto graph = std::make_shared<Graph>();
+  graph->setName("benchmark_conv_fprop");
+  graph->setIODataType(DataType::Half).setComputeDataType(DataType::Float);
+
+  auto X = graph->tensor(TensorAttr()
+                             .setName("image")
+                             .setDim({n, c, h, w})
+                             .setStride({c * h * w, h * w, w, 1})); // NCHW
+
+  auto W = graph->tensor(TensorAttr()
+                             .setName("filter")
+                             .setDim({k, c, r, s})
+                             .setStride({c * r * s, r * s, s, 1})); // KCRS
+
+  auto conv_attr = ConvFPropAttr()
+                       .setStride({u, v})
+                       .setPadding({p, q})
+                       .setDilation({l, j})
+                       .setName("conv_fprop");
+
+  auto Y = graph->convFProp(X, W, conv_attr);
+  Y->setOutput(true);
+
+  // Validate, infer missing properties
+  FUSILLI_CHECK_ERROR(graph->validate());
+
+  // Compile
+  FUSILLI_CHECK_ERROR(graph->compile(handle, /*remove=*/true));
+
+  // Allocate input buffer.
+  auto xBuf = std::make_shared<Buffer>(FUSILLI_TRY(
+      Buffer::allocate(handle,
+                       /*shape=*/castToSizeT({n, c, h, w}),
+                       /*data=*/std::vector<half>(n * c * h * w, half(1.0f)))));
+
+  // Allocate weight buffer.
+  auto wBuf = std::make_shared<Buffer>(FUSILLI_TRY(
+      Buffer::allocate(handle,
+                       /*shape=*/castToSizeT({k, c, r, s}),
+                       /*data=*/std::vector<half>(k * c * r * s, half(1.0f)))));
+
+  // Allocate output buffer.
+  auto yBuf = std::make_shared<Buffer>(FUSILLI_TRY(
+      Buffer::allocate(handle,
+                       /*shape=*/castToSizeT({n, k, h, w}),
+                       /*data=*/std::vector<half>(n * k * h * w, half(0.0f)))));
+
+  // Create variant pack.
+  const std::unordered_map<std::shared_ptr<TensorAttr>, std::shared_ptr<Buffer>>
+      variantPack = {
+          {X, xBuf},
+          {W, wBuf},
+          {Y, yBuf},
+      };
+
+  // Execute graph a few times.
+  for (size_t i = 0; i < 100; i++)
+    FUSILLI_CHECK_ERROR(graph->execute(variantPack));
+
+  return ok();
+}
+
+int main(int argc, char **argv) {
+  CLI::App mainApp{"Fusilli Benchmark Driver"};
+  mainApp.require_subcommand(1);
+
+  // Conv flags are kept in sync with MIOpen's ConvDriver:
+  // https://github.com/ROCm/rocm-libraries/blob/db0544fb61f2c7bd5a86dce98d4963420c1c741a/projects/miopen/driver/conv_driver.hpp#L878
+  CLI::App *convApp =
+      mainApp.add_subcommand("conv", "Fusilli Benchmark Forward Convolution");
+  int64_t n = 100, c = 3, h = 32, w = 32, k = 32, r = 3, s = 3, u = 1, v = 1,
```

Comment:
Is this the max value? Do we need to even specify this? Quick scan of things, we should probably make all of the required?

Reply:
These are defaults from MIOpen's driver: https://github.com/ROCm/rocm-libraries/blob/develop/projects/miopen/driver/conv_driver.hpp#L915
So we're at parity with it when invoked without any args. They can be made required as well if needed.

Reply:
Removed defaults and made all options required: https://github.com/nod-ai/shark-ai/pull/2410/commits/402c6258c079cb51db208c8d28746637ec600d39

Also added range checks on options.